plugins/gatsby-theme-iterative-docs/src/components/Documentation/Markdown/index.tsx

```diff
@@ -38,18 +43,44 @@ const Details: React.FC<Record<string, never>> = ({ children }) => {
     0,
     firstChild.props.children.length - 1
   ) as ReactNode[]
+  const id = triggerChildren
```

Comment:
Would it be possible to use `github-slugger` to create the anchor links? That could fix the issue about details blocks having the same title 🤔

Reply:
Umm, I tried using `github-slugger` and it doesn't seem to provide any significant improvement. The slug also has encoded values for icons used on titles which don't seem nice. `/doc/start/data-and-model-versioning#%EF%B8%8F-expand-to-get-an-example-dataset`. 
For the issue having the same title, we need to come up with another solution. For now, I could think of two ways but the solution has some drawbacks
1. Generating some random string at the end of this particular title `Expand to see what happens under the hood.
` but the problem would be to make them unique on multiple builds. 
2. Manually setting unique values either on the title or setting some props like numbering the same title. Still, there could be possibility of having same title.

Reply:
2. numbers (`-2`, `-3`, ...) can be appended to `#id`s automatically (rather than manually), right?

Reply:
For now, I cannot think of making it fully automatic because currently, we check from the `<details>`  block from where we cannot know if we have another details block with the same title on a page. Maybe, we can check somewhere from a higher block eg:`page` but need some research on that.

Reply:
> Umm, I tried using github-slugger and it doesn't seem to provide any significant improvement.

No significant improvement? Was it not able to make unique anchor links? 

> The slug also has encoded values for icons used on titles which don't seem nice. /doc/start/data-and-model-versioning#%EF%B8%8F-expand-to-get-an-example-dataset.

We could take off the icons with regex then use the remaining string with `github-slugger`? That is, if `github-slugger` is able to create the desired unique anchor links that we need :)